### PR TITLE
[SW-1112] update install_spot_ros2.sh for SDK 4.0.2

### PR DESCRIPTION
**Pull request by khughes-bdai**
## Change Overview

Updates this repo for spot-sdk 4.0.2!

- [x] The update to spot_wrapper needs to go in first: https://github.com/bdaiinstitute/spot_ros2/pull/401

## Testing Done

- [x] locally ran the install script in my ros2 environment (outside of docker). Verified that installations of bosdyn_msgs, the python bosdyn sdk, and spot-cpp-sdk were all updated to 4.0.2. 
- [x] Tested all `spot_examples` on robot as well as a few standard service calls
- [x] CI

### CODE REVIEW

**Comment by khughes-bdai** [comment]
* **#404** <a href="https://app.graphite.dev/github/pr/bdaiinstitute/spot_ros2/404?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @khughes-bdai and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by tcappellari-bdai — APPROVED** [review]
lgtm

File: install_spot_ros2.sh [hunk 1]
```diff
@@ -38,6 +38,6 @@ yes | sudo /tmp/ros-humble-bosdyn_msgs_${MSG_VERSION}-jammy_${ARCH}.run  --nox11
 rm /tmp/ros-humble-bosdyn_msgs_${MSG_VERSION}-jammy_${ARCH}.run
 
 # Install spot-cpp-sdk
-wget -q -O /tmp/spot-cpp-sdk_${SDK_VERSION}_${ARCH}.deb https://github.com/bdaiinstitute/spot-cpp-sdk/releases/download/${SDK_VERSION}/spot-cpp-sdk_${SDK_VERSION}_${ARCH}.deb
+wget -q -O /tmp/spot-cpp-sdk_${SDK_VERSION}_${ARCH}.deb https://github.com/bdaiinstitute/spot-cpp-sdk/releases/download/v${SDK_VERSION}/spot-cpp-sdk_${SDK_VERSION}_${ARCH}.deb
```
[the remark this answers is not in the document]
Comment (mhidalgo-bdai) [reply install_spot_ros2.sh:41]: We changed the tagging pattern --> https://github.com/bdaiinstitute/spot-cpp-sdk/tags

**Review by mhidalgo-bdai — APPROVED** [review]

File: install_spot_ros2.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (amessing-bdai) [reply install_spot_ros2.sh:41]: That was not. I just didn't remember that BD created the tags with `v` and we generated the ones without it

[the remark this answers is not in the document]
Comment (khughes-bdai) [reply install_spot_ros2.sh:41]: I'm fine with leaving it with the `v`, just wanted to bring it up

**Review by bhung-bdai — APPROVED** [review]

**Pull request merged**